pypodcastparser/Item.py

```diff
@@ -208,15 +226,74 @@ def set_guid(self, tag):
         except AttributeError:
             self.guid = None
 
+
+#TODO convert to one timezone
     def set_published_date(self, tag):
         """Parses published date and set value."""
         try:
             self.published_date = tag.string
             self.published_date_string = tag.string
-            pubdate = parse(self.published_date)
-            self.published_date = datetime.datetime.strftime(pubdate, "%Y-%d-%m, %H:%M:%S")
-        except AttributeError:
-            self.published_date = None
+
+            deconstructed_date = self.published_date_string.split(" ")
+            if(len(deconstructed_date) < 4):
+                raise AttributeError
+
+            published_date_timezone=""
+            if (re.match("^[a-zA-Z]{3}$", deconstructed_date[-1])):
+                published_date_timezone= deconstructed_date[-1]
+                deconstructed_date.pop()
+            else:
+                published_date_timezone= 'EST'
+
+
+            regex_array = ["^[a-zA-Z]{3},$","^\d{1,2}$","^[a-zA-Z]{3}$","^\d{4}$","^\d\d:\d\d"]
+            new_array = []
+            for array_index, array_value in enumerate(regex_array):
+                if(re.match(deconstructed_date[array_index],array_value)):
+                    new_array.append(array_value)
+                else:
+                    for inner_index,inner_value in enumerate(deconstructed_date):
+                        if(re.match(regex_array[array_index],inner_value)):
+                            new_array.append(inner_value)
+                            break
+            date_string = new_array[0]+" "+new_array[1]+" "+ new_array[2]+" "+new_array[3]+" "+new_array[4]
+
+            if(len(new_array) != 5):
+                raise AttributeError
```

Comment:
Can we add a comment onto this AttributeError?  Maybe something like "length of date array is incorrect expected 5 but got x"

Reply:
added error message